packages/suite/src/hooks/suite/usePinModal.ts

```diff
@@ -0,0 +1,28 @@
+import { selectDeviceButtonRequests } from '@suite-common/wallet-core';
+import { useSelector } from 'src/hooks/suite';
+
+export const usePin = () => {
+    const buttonRequests = useSelector(selectDeviceButtonRequests);
+
+    const pinRequestType = buttonRequests[buttonRequests.length - 1];
+    const invalidCounter = buttonRequests.filter(r => r.code === 'ui-invalid_pin').length || 0;
+
+    const isExtended =
```

Comment:
why  `isExtended`? would be better to name: `isRequestingNewPinCode`?

Reply:
See the comment:

```
// 3 cases when we want to show left column
// 1) and 2) - Setting a new pin: 1 entry, 2nd (confirmation) entry
// 3) Invalid pin (It doesn't seem to work anymore) instead separate PinMismatchModal is shown
```

Point 3 applies to an existing PIN, so the original naming is actually more apt. Could be changed to `isModalExtended` or something for extra clarity.

Reply:
Thanks, good point! Now I also understand the comment. 

So suggest to do this them:
```ts
const isRequestingNewPinCode = (pinRequestType?.code && newPinRequestTypes.includes(pinRequestType.code))

const isPinInvalid = invalidCounter > 0; // maybee better naming, I still dont understand when is pin invalid in the settings (too short, too weak? is it this type of validation?

const isModalExtended = isRequestingNewPinCode || isPinInvalid
```